### ISSUE TITLE
uncle reward

### ISSUE THREAD

**Issue by tapouillo** [sammy007/open-ethereum-pool#311]
Hi,

we are getting an incorrect block uncle reward (on ETC).
everything works well on other blocks (around 4 ETC), but the uncle reward should be 0.125, not 3.
I'm unable to find the correct parameter to change this ?

block:
https://gastracker.io/uncle/0x8a69032e6bc13e1a3dd9f11af17a955087dec5e20b0032e023e3b892f1b821d9

result, is that 3 etc is shared to all miners, but the pool received only 0.125.

how to fix this ?
(I also tried the lechuckDE fork, but this one is even unable to unlock the blocks (wrong gas format error).

#### System information

Geth
Version: v5.0.0-ac28808
Protocol Versions: [63 62]
Network Id: 1
Go Version: go1.9.4
OS: linux

Redis version: `redis-server --version`
Redis server v=3.2.10 sha=00000000:0 malloc=jemalloc-3.6.0 bits=64 build=c8b45a0ec7dc67c6

OS & Version: Linux/OSX
centos7

Commit branch and hash: (`master`, `bcfc0eb`)

**Comment by mikeyb** [comment]
If you want this to work for ETC network, modify the source.  This is written for ETH network.  Use https://github.com/LeChuckDE/open-ethereumclassic-pool or update the code for your needs, this is not anyone elses job but YOURS

**Comment by tapouillo** [comment]
Nice constructive  answer. If you read my question, you would have seen that I already have modified the source, that I already tried the other fork (which as problems) and that this one works as expected , except a little problem with the uncle rewards .

I also would like to mention that Sammy007 answered in another thread issue that we can use this to run etc like eth.

Any help would be appreciated.

**Comment by mikeyb** [comment]
If you read your question you will see you say you are using the master branch, no mention of modifications, and if you also read the massive amount of shit posts asking for HELP in a BUG TRACKER, you would realize my answer is about as constructive as you can get

To help you:
1.  You might want to double check the uncle reward on ETC, I am betting you missed a key number
2.  It sounds like it is running just fine to me

**Comment by tapouillo** [comment]
I had no doubt I would get such constructive answer from you

**Comment by mikeyb** [comment]
Since you are so adament on not reading.  THAT IS THE UNCLE REWARD, NOTHING IS WRONG!!!

**Comment by yozanisak** [comment]
@mikeyb stop yelling at people maybe you will understand better the question. reading the other issues, looks like it's your favorite sport... pathetic .  if you have the answer, instead of trying to give lessons, just help the community, that's what people usually do here even if it's called 'issue tracker'. 

@tapouillo the ETH uncle calc is (U_n + 8 - B_n) * R / 8  , which is probably why your open-ethereum-pool is reporting 3 (and not the 0.125 you actually got). For ETC: https://medium.com/@cseberino/calculating-ethereum-classic-mining-rewards-with-the-new-monetary-policy-9a1eccdfeb4c,  it is said 0.125 per uncle block. Which seems to correspond to what you got.  

I'm not sure about this, but even the fork https://github.com/LeChuckDE/open-ethereumclassic-pool is using the ETH calc, check the payout/unlocker.go. func handleUncle and getUncleReward. Maybe you can change the getUncleReward to return 0.125 ? Anyone could confirm this?

**Comment by bobbieltd** [comment]
@yozanisak Thank you alot for your help. New monetary policy is launched in December 2017. I’ll take a look at codes.

**Comment by bobbieltd** [comment]
@mikeyb Could you calm down and stop attack others so quickly ? I got it : it is for Ethereum, it’s for bug, it’s not help desk. What is purpose of publishing a repo only for bug finding ? Any bounty 😂

**Comment by mikeyb** [comment]
@yozanisak NO!  Idiots come here trying to run a monetary service to the world, but are too stupid to read.  So stop being so stupid and I will stop being such an asshole.

@bobbieltd NO!  Idiots come here trying to run a monetary service to the world, but are too stupid to read.  So stop being so stupid and I will stop being such an asshole.

**Comment by mikeyb** [comment]
And the purpose of publishing a open source repo is so others can USE the software.  NOT bug the person who created it how to be not stupid.  There is no bounty, it is OPEN SOURCE.  You have to READ and stuff.  I realize this is hard if facebook doesn't tell you how to live your life, but once in a while you gotta use your own damn mind and process logical thoughts.  I won't apologize or act differently to cater to a world of idiots.

**Comment by tapouillo** [comment]
thanks a lot @yozanisak , that make sens, I made the changed and will let you know the result once we get another uncle...

**Comment by defari** [comment]
@tapouillo hi, may i know..did you solve this issue?

**Comment by tapouillo** [comment]
@defari I believe so,  but I can't tell you yet, we didn't get another uncle. I will let you know know.

**Comment by Locky7** [comment]
@tapouillo Did it work?

**Comment by tapouillo** [comment]
unfortunately our group of miners moved to another coin before having found another ETC uncle. I can't actually confirm but that should work, that's where the uncle reward is calculated

**Comment by Rem559** [comment]
for sammy007/open-ethereum-pool repo put the function getUncleReward in the file unloker.go to the following form:
func getUncleReward(uHeight, height int64) *big.Int {
	reward := getConstReward(height)
	reward.Div(reward, big.NewInt(32))
	return reward
}
my pool - rca-pool.ru

**Comment by Rem559** [comment]
who helped this solution can throw me a beer
ETC 0x1a59cc767ba3098498f115bc44f2a10456384bb3
